L1Trigger/L1TTrackMatch/plugins/L1TkHTMissEmulatorProducer.cc

```diff
@@ -0,0 +1,286 @@
+    LogError("TkHTMissEmulatorProducer") << "\nWarning: TkJetCollection not found in the event. Exit\n";
+    return;
+  }
+
+  if (!L1TkJetsHandle.isValid() && displaced_) {
+    LogError("TkHTMissEmulatorProducer") << "\nWarning: TkJetExtendedCollection not found in the event. Exit\n";
+    return;
+  }
+
+  int cosLUTbins = floor(l1tmhtemu::kMaxCosLUTPhi / l1tmhtemu::kStepPhi);
+
+  // floats used for debugging
+  float sumPx_ = 0;
+  float sumPy_ = 0;
+  float HT_ = 0;
+
+  float sumPxtemp_ = 0;
+  float sumPytemp_ = 0;
+  float HTtemp_ = 0;
+
+  l1tmhtemu::Et_t sumPx = 0;
+  l1tmhtemu::Et_t sumPy = 0;
+  l1tmhtemu::Et_t HT = 0;
+
+  // loop over jets
+  int jetn = 0;
+  int jetnpasscuts = 0;
+  for (jetIter = L1TkJetsHandle->begin(); jetIter != L1TkJetsHandle->end(); ++jetIter) {
+
+    // floats used for debugging
+    float tmp_jet_px_ = jetIter->pt()*cos(jetIter->glbphi());
+    float tmp_jet_py_ = jetIter->pt()*sin(jetIter->glbphi());
+    float tmp_jet_et_ = jetIter->pt(); // FIXME Get Et from the emulated jets
+    float tmp_jet_pt_ = jetIter->pt();
+    
+
+    l1tmhtemu::pt_t tmp_jet_pt = l1tmhtemu::digitizeSignedValue<l1tmhtemu::pt_t>(jetIter->pt(), l1tmhtemu::kInternalPtWidth, l1tmhtemu::kStepPt);
+    l1tmhtemu::eta_t tmp_jet_eta = l1tmhtemu::digitizeSignedValue<l1tmhtemu::eta_t>(jetIter->glbeta(), l1tmhtemu::kInternalEtaWidth, l1tmhtemu::kStepEta);
+    l1tmhtemu::phi_t tmp_jet_phi = l1tmhtemu::digitizeSignedValue<l1tmhtemu::phi_t>(jetIter->glbphi(), l1tmhtemu::kInternalPhiWidth, l1tmhtemu::kStepPhi);
+    l1tmhtemu::ntracks_t tmp_jet_nt = l1tmhtemu::ntracks_t(jetIter->nt());
+
+    l1tmhtemu::phi_t tmp_jet_cos_phi = l1tmhtemu::phi_t(-999);
+    l1tmhtemu::phi_t tmp_jet_sin_phi = l1tmhtemu::phi_t(-999);
+
+
+    if(tmp_jet_phi >= 0){
+      tmp_jet_cos_phi = cosLUT_[tmp_jet_phi];
+
+      if(cosLUTbins/2 + 1 - tmp_jet_phi >= 0){
+	tmp_jet_sin_phi = cosLUT_[cosLUTbins/2 + 1 - tmp_jet_phi];
+      }
+      else{
+	tmp_jet_sin_phi = cosLUT_[-1 * (cosLUTbins/2 + 1 - tmp_jet_phi)];
+	}
+
+    }
+    else{
+      tmp_jet_cos_phi = cosLUT_[-1*tmp_jet_phi];
+
+      if(cosLUTbins/2 + 1 - (-1*tmp_jet_phi) >= 0){
+	tmp_jet_sin_phi = -1*cosLUT_[cosLUTbins/2 + 1 - (-1*tmp_jet_phi)];
+      }
+      else{
+	tmp_jet_sin_phi = -1*cosLUT_[-1 * (cosLUTbins/2 + 1 - (-1*tmp_jet_phi))];
+	}
+    }
+
+    l1tmhtemu::Et_t tmp_jet_px = tmp_jet_pt*tmp_jet_cos_phi;
+    l1tmhtemu::Et_t tmp_jet_py = tmp_jet_pt*tmp_jet_sin_phi;
+
+    jetn++;
+
+
+    if(debug_){
+      edm::LogVerbatim("L1TrackerHTMissEmulatorProducer")
+      << "****JET EMULATION"<< jetn << "****\n"
+      << "FLOATS ORIGINAL\n" << "PT: " << jetIter->pt() << "| ETA: " << jetIter->glbeta() << "| PHI: " << jetIter->glbphi() << "| NTRACKS: " << jetIter->nt()<< "| COS(PHI): " << cos(jetIter->glbphi()) << "| SIN(PHI): "<< sin(jetIter->glbphi())<< "| Px: " << jetIter->pt()*cos(jetIter->glbphi()) << "| Py: " << jetIter->pt()*sin(jetIter->glbphi())<< "\n"
```

Comment:
These are really long lines. You can break up the lines into shorter pieces and this will be more readable.
This is recommended as per rule 28 of https://cms-sw.github.io/cms_coding_rules.html#4--technical-coding-rules-1.

Reply:
broke up into shorter pieces. I can break it more if you think it can be improved.

Reply:
My general rule of thumb is to keep lines under 120 characters, if possible. It's more of a personal rule than a CMS mandated one. So you can use that if you think it helps. Not sure what you have now. It does look better.